pkg/manager/leaderelection.go

```diff
@@ -41,34 +41,34 @@ func (k *KubeMacPoolManager) waitToStartLeading(poolManger *poolmanager.PoolMana
 }
 
 // Adds the leader label to elected pod and removes it from all the other pods, if exists
-func (k *KubeMacPoolManager) UpdateLeaderLabel() error {
+func (k *KubeMacPoolManager) UpdateLeaderLabel(kubeClient client.Client) error {
```

Comment:
We don't need method receiver anymore since `k` is not used also this function can be private it's only used at this package.

Reply:
well it's used by the test so I think I can't make it private

Reply:
Test is insde the package you can make it private